mtraining/src/main/java/org/motechproject/mtraining/dto/CourseUnitDeserializer.java

```diff
@@ -0,0 +1,92 @@
+package org.motechproject.mtraining.dto;
+
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.JsonProcessingException;
+import org.codehaus.jackson.ObjectCodec;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.type.TypeFactory;
+import org.codehaus.jackson.type.JavaType;
+import org.motechproject.mtraining.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@code CourseUnitDeserializer} for the {@code List<CourseUnitDto>}.
+ */
+public class CourseUnitDeserializer extends JsonDeserializer<List<CourseUnitDto>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CourseUnitDeserializer.class);
+
+    private ObjectMapper mapper;
+    private List<CourseUnitDto> units;
+    private JsonNode jsonNode;
```

Comment:
why are all of the above class fields, if we reinitialize them with every method call? Cant they be mehod local variables?